packages/js/src/dashboard/components/unsaved-changes-modal.js

```diff
@@ -0,0 +1,56 @@
+import { __ } from "@wordpress/i18n";
+import { useCallback } from "@wordpress/element";
+import { get } from "lodash";
+import { useBlocker } from "react-router-dom";
+import { Modal, Button, useSvgAria } from "@yoast/ui-library";
+import { ExclamationIcon } from "@heroicons/react/outline";
+
+/**
+ * The unsaved changes modal.
+ *
+ * @returns {JSX.Element} The unsaved changes modal.
+ */
+export const UnsavedChangesModal = () => {
+	const svgAriaProps = useSvgAria();
+
+	const blocker = useBlocker( ( { currentLocation, nextLocation } ) =>{
+		const isStepBeingEdited = get( window, "isStepBeingEdited", false );
+		return isStepBeingEdited && currentLocation.pathname === "/first-time-configuration" && nextLocation.pathname !== "/first-time-configuration";
+	} );
+
+	const dismiss = useCallback( () => {
+		blocker.reset();
+	}, [ blocker ] );
+
+	const leave = useCallback( () => {
+		blocker.proceed();
+	}, [ blocker ] );
```

Comment:
Why not use `block.reset` and `blocker.proceed` directly?

If it is needed for some reason I don't see, the dependencies should be more specific, e.g. `blocker.reset` or `blocker.proceed`